### ISSUE TITLE
twisted.web examples directory explanation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3418]
|<img alt="temoto's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| temoto reported|
|-|-|
|Trac ID|trac#3418|
|Type|enhancement|
|Created|2008-09-07 13:55:24Z|

Add at least the shortest description to the examples located at http://twistedmatrix.com/projects/web/documentation/examples/

at first i thought module docstrings would do nice, but examples don't have docstrings.

In example, it's hard to find that basic example usage of, well, twisted.web.client is in getpage.py.

Attachments:

* [doc-web-examples-index.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/759/7593098f535355a7b4cc172885895c5b439a24ef/01618824cd4f9831eba266574dbfb407638ac8b8.patch) (4759 bytes) - added by jesstess on 2009-08-16 20:51:38Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3418 3418
type__enhancement enhancement
reporter__temoto temoto
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation_examples documentation examples
time__1220795724000000 1220795724000000
changetime__1251214154000000 1251214154000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_1"></a>|@jesstess set owner to @jesstess|
|-|-|

Added a description for each example and grouped them by topic. 3 examples were removed: webmon.py and tkwebmon.py, because it appears that twisted.web.monitor was removed entirely a while ago according to eg ticket 1878 and changeset 25431, and simple.tpl for not actually being an example but making it into the index anyway.

The handful of particularly lame descriptions are lame because it wasn't obvious me how to run those examples usefully; if someone give me better descriptions I'll happily include them and re-submit the patch.

It seems like webmon.py, tkwebmon.py, and simple.tpl should be removed from the examples/ directory - does a separate ticket need to be created for that?

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

This looks good to me.  The descriptions don't even seem all that lame to me. :)  They're descriptive enough to get people going in the right direction, I think (and the grouping also helps with that a lot).

I'm going to apply this patch.  I agree about deleting webmon.py and tkwebmon.py.  And I guess simple.tpl should go too, since it doesn't seem to be used by anything.  I wonder what it used to be for.  Ah, it's a lore template, presumably for the index.  But I'm pretty sure we don't use it anymore, so yea it can go.  I'll just delete them when I apply the attached patch, since this ticket is basically about making the web examples directory more coherent, and getting rid of stuff that is useless or broken will help with that. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [27241]) Apply doc-web-examples-index.patch and delete several broken/useless files

Author: jesstess
Reviewer: exarkun
Fixes: [#3418](https://github.com/twisted/twisted/issues/3418)

Group the web examples logically in the index file and add some
descriptive text for each one to make it easier to find particular
examples.

Also delete the webmon examples, since webmon was deleted a long
time ago, and delete the simple lore template which is no longer
used when generating these docs.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
